app/models/bulk_submission.rb

```diff
@@ -3,6 +3,25 @@ class BulkSubmission < ApplicationRecord
 
   belongs_to :user
   has_many :submissions, dependent: :destroy
+
+  has_many :unfinished_submissions,
+            -> { where.not(status: %w[completed failed exhausted]) },
+            class_name: "Submission"
+
   has_one_attached :original_file
-  has_status :pending, :preparing, :prepared, :processing, :completed
+  has_one_attached :result_file
+
+  has_status :pending,
+             :preparing,
+             :prepared,
+             :processing,
+             :completed,
+             :writing,
+             :exhausted,
+             :ready
```

Comment:
status seems to be missing `failed`

Reply:
that status is only for submissions, not bulk_submissions, and is one returned by hmrcinterface